File: back-end/src/main/java/com/tppartdeux/edoctorat/dto/candidat/CandidatResponse.java
package com.tppartdeux.edoctorat.dto.candidat;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Candidat response DTO matching Angular's Candidat interface
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CandidatResponse {
    private Long id;
    private String cne;
    private String pays;
    private String nom;
    private String prenom;
    private String email;
    private String cin;
    private String nomCandidatAr;
    private String prenomCandidatAr;
    private String adresse;
    private String adresseAr;
    private String sexe;
    private String villeDeNaissance;
    private String villeDeNaissanceAr;
    private String ville;
    private String dateDeNaissance;
    private String typeDeHandiCape;
    private String academie;
    private String telCandidat;
    private String pathCv;
    private String pathPhoto;
    private Integer etatDossier;
    private String commentaireScolarite;
    
    @JsonProperty("situation_familiale")
    private String situationFamiliale;
    
    private String fonctionnaire;
}

File: front-end/src/models/CandidatResponse.ts
export interface CandidatResponse {
    id: number;
    cne: string;
    pays: string;
    nom: string;
    prenom: string;
    email: string;
    cin: string;
    nomCandidatAr: string | undefined;
    prenomCandidatAr: string | undefined;
    adresse: string;
    adresseAr: string | undefined;
    sexe: string;
    villeDeNaissance: string;
    villeDeNaissanceAr: string | undefined;
    ville: string;
    dateDeNaissance: string;
    typeDeHandiCape: string;
    academie: string | undefined;
    telCandidat: string;
    pathCv: string | undefined;
    pathPhoto: string | undefined;
    etatDossier: number | undefined;
    commentaireScolarite: string | undefined;
    situation_familiale: string | undefined;
    fonctionnaire: string | undefined;
    sujetPostule: string;
    directeurNom: string;
    directeurPrenom: string;
    codirecteurNom: string;       // Peut être null
    codirecteurPrenom: string;    // Peut être null
    formationDoctorale: string;
}
